drivers/ukrandom/lcpu/Config.uk

```diff
@@ -0,0 +1,11 @@
+menuconfig LIBUKRANDOM_LCPU
+	bool "CPU generated randomness"
+	depends on (ARCH_X86_64 || (ARCH_ARM_64))
+	select HAVE_ARM64_FEAT_RNG if ARCH_ARM_64
```

Comment:
So in the Arm CPU configuration  `HAVE_ARM64_FEAT_RNG` is an invisible option only, that is also selected by features that depend on randomness. It needs to be there because we use it to generate GCC flags: https://github.com/unikraft/unikraft/blob/staging/arch/arm/arm64/Makefile.uk#L68. Ideally I would had liked that these flags are generated per-library, but unfortunately there are some inter-dependencies on the `march` flag that we need to resolve first at the build system. Example: 

* `FEAT_RNG` resolves to ARCH_REV := armv8.5-a and ARCH_OPT := $(ARCH_OPT)+rng
* `FEAT_PAUTH` resolves to ARCH_REV := armv8.3-a and BRANCH_PROTECTION := pac-ret+leaf
 
At the final compiler flags we need to select the highest architectural revision, otherwise we'll get a compiler error that `FEAT_RNG` is not supported in `armv8.3-a`. For now I choose to have libraries select the architectural features they support (e.g. `FEAT_RNG`) and delegate the compiler flags generation to a single Makefile where we easily can do the dependency resolution by specifying features of later architectural revisions last, so that they override earlier ones.

> Shouldn't the random driver depend on HAVE_ARM64_FEAT_RNG?

The reason we can't have `libukrandom` depend on `FEAT_RNG` is that we don't know if the underlying CPU implements `FEAT_RNG`. On `arm64` GCC defines some generic CPUs , i.e. `MCPU_ARM64_NONE`, `MCPU_ARM64_NATIVE` and `MCPU_ARM64_GENERIC`, and the default we use in Kconfig is`MCPU_ARM64_NONE`. So when the user selects the lcpu rng driver of `arm64`, it is expected that the user implies that the CPU implements `FEAT_RNG`.

Reply:
Okay, understand. As we discussed, let's re-visit this integration when we touch the compiler settings as part of the final re-arch of the architecture/lcpu libraries. I am thinking that we may want to rename `HAVE_ARM64_FEAT_RNG` to `NEED_ARM64_FEAT_RNG` or `WANT_ARM64_FEAT_RNG` - but getting to a conclusion is not trivial at this point and needs further discussion. We shouldn't block this PR because of this. Can you add a comment to the source code that describes why we are doing this here (for now) with `HAVE_ARM64_FEAT_RNG`?

Reply:
Done